subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -174,12 +175,16 @@ VirtualFileSystem createVirtualFileSystem(
                     public void afterStart(GradleInternal gradle) {
                         Map<String, String> systemPropertiesArgs = gradle.getStartParameter().getSystemPropertiesArgs();
                         if (isRetentionEnabled(systemPropertiesArgs)) {
+                            DeprecationLogger.incubatingFeatureUsed("Virtual File System Retention");
```

Comment:
Should we refer to `SingleMessageLogger` instead? I mean, this is not a deprecation, so...

Reply:
I think we shouldn't use title case here.

Reply:
Done.